### PR TITLE
peer storage cleanup

### PR DESCRIPTION
**Pull request by NingLin-P**
Signed-off-by: linning <linningde25@gmail.com>

### CODE REVIEW

File: kv/raftstore/peer_storage.go [hunk 1]
```diff
@@ -508,10 +439,15 @@ func (ps *PeerStorage) SaveReadyState(ready *raft.Ready) (*ApplySnapResult, erro
 	kvWB, raftWB := new(engine_util.WriteBatch), new(engine_util.WriteBatch)
 	ctx := NewInvokeContext(ps)
```
Comment (Connor1996) [inline kv/raftstore/peer_storage.go]: actually, we can remove the invoke context. Let's leave it to future work

File: kv/raftstore/peer_storage.go [hunk 2]
```diff
@@ -544,26 +473,15 @@ func (ps *PeerStorage) SaveReadyState(ready *raft.Ready) (*ApplySnapResult, erro
...
 	// If we apply snapshot ok, we should update some infos like applied index too.
-	if ctx.SnapRegion == nil {
-		return nil, nil
+	if snapshotIdx > 0 {
+		ps.region = ctx.SnapRegion
+		ps.ScheduleApplyingSnapshot()
 	}
-	// cleanup data before scheduling apply worker.Task
-	if ps.isInitialized() {
-		ps.clearExtraData(ps.region)
```
Comment (Connor1996) [inline kv/raftstore/peer_storage.go]: why remove this

File: kv/raftstore/peer_storage_test.go [hunk 3]
```diff
@@ -194,37 +194,6 @@ func TestPeerStorageEntries(t *testing.T) {
 	}
 }
 
-func TestPeerStorageCompact(t *testing.T) {
```
Comment (Connor1996) [inline kv/raftstore/peer_storage_test.go]: why remove this?

File: kv/raftstore/peer_storage.go (hunk 2 above)
Comment (NingLin-P) [reply kv/raftstore/peer_storage.go]: `clearExtraData` is used to clear data that not overlap by `ps.region` and the input region, but in here the input region is `ps.region` too, so they always overlap and `clearExtraData` is useless.

File: kv/raftstore/peer_storage_test.go (hunk 3 above)
Comment (NingLin-P) [reply kv/raftstore/peer_storage_test.go]: This test case is mainly used to test `CompactRaftLog` which had been removed.

File: kv/raftstore/peer_storage.go (hunk 2 above)
Comment (Connor1996) [reply kv/raftstore/peer_storage.go]: after apply snapshot, the region may change

Comment (NingLin-P) [reply kv/raftstore/peer_storage.go]: oh, got it.

File: kv/raftstore/peer_storage.go (hunk 1 above)
Comment (NingLin-P) [reply kv/raftstore/peer_storage.go]: I had removed it, PTAL.

File: kv/raftstore/peer_storage.go [hunk 4]
```diff
@@ -449,53 +394,49 @@ func ClearMeta(engines *engine_util.Engines, kvWB, raftWB *engine_util.WriteBatc
...
-
-	ctx.ApplyState.AppliedIndex = lastIdx
-	// The snapshot only contains log which index > applied index, so
-	// here the truncate state's (index, term) is in snapshot metadata.
-	ctx.ApplyState.TruncatedState = &rspb.RaftTruncatedState{
-		Index: lastIdx,
-		Term:  snap.Metadata.Term,
+	ps.raftState.LastIndex = snap.Metadata.Index
+	ps.lastTerm = snap.Metadata.Term
+	applyState := &rspb.RaftApplyState{
```
Comment (Connor1996) [inline kv/raftstore/peer_storage.go]: why not call `saveApplyStateTo` here? no need to return it

File: kv/raftstore/peer.go [hunk 5]
```diff
@@ -206,12 +202,6 @@ func (p *peer) MaybeDestroy() bool {
 		log.Infof("%v is being destroyed, skip", p.Tag)
 		return false
 	}
-	if p.IsApplyingSnapshot() {
-		if !p.Store().CancelApplyingSnap() {
```
Comment (Connor1996) [inline kv/raftstore/peer.go]: please keep this now, it's just a place holder. May finish the implementation later

File: kv/raftstore/peer_storage.go [hunk 6]
```diff
@@ -506,64 +447,60 @@ func (ps *PeerStorage) ApplySnapshot(ctx *InvokeContext, snap *eraftpb.Snapshot,
...
 	if len(ready.Entries) != 0 {
-		if err := ps.Append(ctx, ready.Entries, raftWB); err != nil {
+		if err := ps.Append(ready.Entries, raftWB); err != nil {
 			return nil, err
 		}
 	}
 
-	// Last index is 0 means the peer is created from raft message
-	// and has not applied snapshot yet, so skip persistent hard state.
-	if ctx.RaftState.LastIndex > 0 {
```
Comment (Connor1996) [inline kv/raftstore/peer_storage.go]: why remove this

Comment (NingLin-P) [reply kv/raftstore/peer_storage.go]: Seems it just a optimization only works one time, but the comment is quite confusing as the `RaftInitLogIndex = 5`.

**Pull request merged**